### ISSUE TITLE
[web]: Syncing files to device Chrome - Stuck

### ISSUE THREAD

**Issue by Dibitybopty** [flutter/flutter#54290]
Hi. I've had this issue when in the master channel of flutter for a couple of weeks now, where sometimes the 'Syncing files to device Chrome' dialog wouldn't go away in VSCode. The issue wasn't in the beta channel until today and is still in the master channel.

It seems to happen at a complete random save (hot reload) and I have to refresh the browser to make it go away. The strange thing is that it actually loads the data with no issue, but I can't hot reload until I refresh the browser or restart the device.

I'm using the latest version of Chrome (Google Chrome is up to date
Version 81.0.4044.92 (Official Build) (64-bit))

```console
E:\xampp\htdocs\phplessons\ems_new>flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[ÔêÜ] Flutter (Channel master, v1.18.1-pre.26, on Microsoft Windows [Version 6.1.7601], locale en-GB)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses
[ÔêÜ] Chrome - develop for the web
[ÔêÜ] Android Studio (version 3.4)
[ÔêÜ] Connected device (2 available)

! Doctor found issues in 1 category.''

```

```console
>Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, v1.17.0, on Microsoft Windows [Version 6.1.7601], locale en-GB)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[√] Chrome - develop for the web
[√] Android Studio (version 3.4)
[√] Connected device (2 available)

>! Doctor found issues in 1 category.
```

**Comment by iapicca** [comment]
Hi @Dibitybopty 
Does it happen only on VS Code or also from terminal?
can you please provide your `flutter run -d --verbose` 
and a reproducible minimal code sample.
Thank you

**Comment by Dibitybopty** [comment]
Hi @iapicca 
I can confirm that it also happens from the terminal, it's currently stuck now at 'performing hot restart...' and hitting 'r' is unresponsive.
The only thing I changed was adding and removing the alignment property in a container that was inside a slidable widget. Though it will happen with almost any type of change at what seems to be random times.

What I can say is that it will won't get stuck until quite a few hot reloads, but once it happens and I refresh the browser to get rid of it, it will happen very often - nearly every other hot restart.

**Comment by TahaTesser** [comment]
Hi @Dibitybopty 
Are you still experiencing this issue?
Can you please confirm wit latest master?
Thank you

**Comment by Dibitybopty** [comment]
Hey @TahaTesser 
Unfortunately it's still happening with the latest master. I'm currently on a new project with only a few widgets and it's very frequent. I've also switched to Android Studio and it's made no difference.

> Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, v1.18.0-6.0.pre.42, on Microsoft Windows [Version 6.1.7601], locale en-GB)
[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[√] Chrome - develop for the web
[√] Android Studio (version 3.6)
[√] Connected device (2 available)

>! Doctor found issues in 1 category.

Current project:

```
void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        debugShowCheckedModeBanner: false,
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
          visualDensity: VisualDensity.adaptivePlatformDensity,
        ),
        home: Scaffold(

          appBar: AppBar(

            title: Text('Hello!'),

          ),
          body: Container(
            child: NavBar()
          ),
        ),
      );
  }
}

import 'package:flutter/material.dart';

class NavBar extends StatefulWidget {
  @override
  _NavBarState createState() => _NavBarState();
}

class _NavBarState extends State<NavBar> {
  List<String> _pages = [
    'HOME',
    'Nav2',
    'Nav3',
  ];

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Align(
      alignment: Alignment.center,
      child: Container(
        child: NavItems(
          pages: _pages,
        ),
      ),
    );
  }
}

class NavItems extends StatefulWidget {
  final pages;

  const NavItems({Key key, @required this.pages}) : super(key: key);

  @override
  _NavItemsState createState() => _NavItemsState();
}

class _NavItemsState extends State<NavItems> {
  int activeIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        for(var i = 0; i < 5;i++)
        GestureDetector(
          onTap: () {
            setState(() {
              activeIndex = i;
            });
          },
          child: Container(
            padding: EdgeInsets.symmetric(vertical: 20, horizontal: 30),
            child: Text(
              '${widget.pages[i]}',
              style: TextStyle(
                  fontSize: 32,
                  fontWeight: FontWeight.bold,
                  color: activeIndex == i ? Colors.black : Colors.blueGrey),
            ),
          ),
        ),
      ],
    );
  }
}
```
Thanks

**Comment by TahaTesser** [comment]
Could be related [#54876](https://github.com/flutter/flutter/issues/54876)

**Comment by mahdisml** [comment]
i have this issue every time , not random times :(

**Comment by zeppelin-tt** [comment]
flutter run -d --verbose get me on flutter beta channel

#0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
#1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:336:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:702:11)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:619:33)
<asynchronous suspension>
#4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart)
#5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#6      _rootRun (dart:async/zone.dart:1184:13)
#7      _CustomZone.run (dart:async/zone.dart:1077:19)
#8      _runZoned (dart:async/zone.dart:1619:10)
#9      runZoned (dart:async/zone.dart:1539:10)
#10     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#11     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:609:20)
#12     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#13     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:339:21)
<asynchronous suspension>
#14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart)
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#16     _rootRun (dart:async/zone.dart:1184:13)
#17     _CustomZone.run (dart:async/zone.dart:1077:19)
#18     _runZoned (dart:async/zone.dart:1619:10)
#19     runZoned (dart:async/zone.dart:1539:10)
#20     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#21     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:288:19)
#22     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#23     new Future.sync (dart:async/future.dart:224:31)
#24     CommandRunner.run (package:args/command_runner.dart:112:14)
#25     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:231:18)
#26     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
#27     _rootRun (dart:async/zone.dart:1184:13)
#28     _CustomZone.run (dart:async/zone.dart:1077:19)
#29     _runZoned (dart:async/zone.dart:1619:10)
#30     runZonedGuarded (dart:async/zone.dart:1608:12)
#31     runZoned (dart:async/zone.dart:1536:12)
#32     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#33     run.<anonymous closure> (package:flutter_tools/runner.dart)
#34     runInContext.runnerWrapper (package:flutter_tools/src/context_runner.dart:64:18)
<asynchronous suspension>
#35     runInContext.runnerWrapper (package:flutter_tools/src/context_runner.dart)
#36     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#37     _rootRun (dart:async/zone.dart:1184:13)
#38     _CustomZone.run (dart:async/zone.dart:1077:19)
#39     _runZoned (dart:async/zone.dart:1619:10)
#40     runZoned (dart:async/zone.dart:1539:10)
#41     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#42     runInContext (package:flutter_tools/src/context_runner.dart:67:24)
#43     run (package:flutter_tools/runner.dart:48:10)
#44     main (package:flutter_tools/executable.dart:69:9)
#45     main (file:///c:/dev/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#46     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#47     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)

Fluter doctor: 

[√] Flutter (Channel beta, v1.17.0-3.2.pre, on Microsoft Windows [Version 10.0.18362.778], locale ru-RU)
    • Flutter version 1.17.0-3.2.pre at c:\dev\flutter
    • Framework revision 2a7bc389f2 (6 days ago), 2020-04-21 20:34:20 -0700
    • Engine revision 4c8c31f591
    • Dart version 2.8.0 (build 2.8.0-dev.20.10)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\maxim\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] IntelliJ IDEA Ultimate Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2019.1.2
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.6494.35

[√] VS Code (version 1.39.1)
    • VS Code at C:\Users\maxim\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.5.1

[√] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.122
    • Web Server • web-server • web-javascript • Flutter Tools

**Comment by mahdisml** [comment]
i solved it with starting a new project with new version of flutter (master channel) (updating isn't enough you must start a new project with it) ! i dont know why but problem is gone .

**Comment by zeppelin-tt** [comment]
> i solved it with starting a new project with new version of flutter (master channel) (updating isn't enough you must start a new project with it) ! i dont know why but problem is gone .

for a while the problem goes away, but do not rush to rejoice. She is coming back very soon. ((

**Comment by jonahwilliams** [comment]
When the hot reload gets stuck, are there any errors in the browser console tab?

**Comment by jonahwilliams** [comment]
FYI @grouma

**Comment by grouma** [comment]
This could be related to https://github.com/dart-lang/webdev/issues/947

I've yet to root cause the issue as it has been on the back burner. We are likely stuck in one of two places:

- [Waiting for the breakpoints to clear](https://github.com/dart-lang/webdev/blob/0a0175eb8086c21670ff8d455338c6d9b5822ab5/dwds/lib/src/dwds_vm_client.dart#L118)
- [Waiting for the isolate to resume](https://github.com/dart-lang/webdev/blob/0a0175eb8086c21670ff8d455338c6d9b5822ab5/dwds/lib/src/dwds_vm_client.dart#L83)

I have a theory that it is related to the keep alive logic in `package:sse`. @DanTup recently added shutdown support to SSE connections which may come in handy to prevent various race conditions,. 

Any console logs from the browser that folks can provided will be greatly appreciated.

**Comment by Dibitybopty** [comment]
> 
> 
> When the hot reload gets stuck, are there any errors in the browser console tab?

@jonahwilliams I'm getting no errors in the console. Nothing but my debug prints.

**Comment by sndrox** [comment]
A workaround that works most of the time is, when the "Syncing files..." is displayed:

Hit reload on Chrome and, once the refresh is finished, press the "Flutter Hot Reload" button on Android Studio.

Looks like this awakens it all and keeps working as it should.

**Comment by grouma** [comment]
I dug into this today. I know what's causing the issue but I don't know the exact root cause.

We are getting a seemingly random injected client disconnect here:
https://github.com/dart-lang/webdev/blob/e6491c5826e89d83b0fd3a6db732e920bc02e1ce/dwds/lib/src/handlers/dev_handler.dart#L241

This causes a destroy isolate:
https://github.com/dart-lang/webdev/blob/e6491c5826e89d83b0fd3a6db732e920bc02e1ce/dwds/lib/src/handlers/dev_handler.dart#L250

As a result we hit:
https://github.com/dart-lang/webdev/issues/947

This seems to only happen after a refresh. I'm thinking it's related to the keep alive logic in `package:sse`. I'll keep digging.

**Comment by grouma** [comment]
@jonahwilliams you'll need to depend on `package:dwds` version `3.1.0` which I just published.

**Comment by jonahwilliams** [comment]
Perfect, thanks @grouma !

**Comment by HvTechnologies** [comment]
hey....i'm still experiencing this issue....can u just explain how to remove it

**Comment by ph55** [comment]
Same here. Issue still exists both on beta and master channels.

**Comment by bsutton** [comment]
I'm also experiencing this on the latest beta.

flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.18.0-11.1.pre, on Linux, locale en_AU.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[✓] Connected device (2 available)

• No issues found!

**Comment by brianorwhatever** [comment]
Still experiencing this with dwds 3.1.1

**Comment by christian-soho** [comment]
I'm also getting stuck on syncing files.

flutter doctor
[√] Flutter (Channel beta, 1.18.0-11.1.pre, on Microsoft Windows [Version 10.0.18362.657], locale en-US)
 
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Chrome - develop for the web
[√] Android Studio (version 3.4)
[√] VS Code (version 1.45.1)
[√] Connected device (2 available)

**Comment by christian-soho** [comment]
> I'm also getting stuck on syncing files.
> 
> flutter doctor
> [√] Flutter (Channel beta, 1.18.0-11.1.pre, on Microsoft Windows [Version 10.0.18362.657], locale en-US)
> 
> [√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
> [√] Chrome - develop for the web
> [√] Android Studio (version 3.4)
> [√] VS Code (version 1.45.1)
> [√] Connected device (2 available)

I've managed to run my project through the terminal with 
`flutter run -d "chrome" --verbose`

**Comment by MyHouseIsBig** [comment]
I have the same problem,

I also tried to run my project trough the terminal with
`flutter run -d chrome --verbose`

the strange thing that happens when i run that command is that it Syncs successfully all the files (at least that's what the --verbose states) but the browser doesn't show any content.

**Comment by grouma** [comment]
cc @DanTup

**Comment by yasinarik** [comment]
I am frustrated by this :( Still happening to me as well. Flutter version v1.15.3 was really good with working.

I am not considering to downgrade, I'll stick with the latest but... 

Thank you all for working on this issue anyways.

**Comment by alexda12** [comment]
Same issue - Stuck on 'Synching files to device Chrome' - white screen shown in chrome and no errors in dart console.

Brand new project (Counter) WORKS. No issue.

Existing app that you : 
a. Flutter web enabled
b. Flutter create .
c. Create index.html as shown below : 

```
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="A new Flutter project.">

  <!-- iOS meta tags & icons -->
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta name="apple-mobile-web-app-title" content="web1">
  <link rel="apple-touch-icon" href="icons/Icon-192.png">

  <!-- Favicon -->
  <link rel="shortcut icon" type="image/png" href="favicon.png"/>

  <title>web1</title>
  <link rel="manifest" href="manifest.json">
</head>
<body>
  <!-- This script installs service_worker.js to provide PWA functionality to
       application. For more information, see:
       https://developers.google.com/web/fundamentals/primers/service-workers -->
  <script>
    if ('serviceWorker' in navigator) {
      window.addEventListener('load', function () {
        navigator.serviceWorker.register('flutter_service_worker.js');
      });
    }
  </script>
<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-app.js"></script>
 	
<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-analytics.js"></script>
<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-auth.js"></script>
<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-storage.js"></script>
<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-remote-config.js"></script>
<script src="./init.js"/>

<script src="main.dart.js" type="application/javascript"></script>
</body>
</html>

```
and my Firebase init code : 
```

var firebaseConfig = {
  apiKey: "my-api-key",
  authDomain: "my-project.firebaseapp.com",
  databaseURL: "https://my-project.firebaseio.com",
  projectId: "my-project",
  storageBucket: "my-project-bucket.appspot.com",
  messagingSenderId: "my-sender-id",
  appId: "my-app-id",
  measurementId: "my-measurement-id",
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);
firebase.analytics();

```
What I've tried : 

> Running from debug - Gets stuck on 'Syncing file to device ...'


1- Latest Beta
2- Latest Dev build (1.19.0-1.0.pre)
3- Latest Master build (1.19.0-2.0.pre.145)
4- Latest chrome 83.0.4103.61
5- Running in Debug (Issue)
6- Running NOT IN DEBUG (Issue)

Running in terminal : 

`Flutter run -d chrome`

```
Launching lib/main.dart on Chrome in debug mode...
Syncing files to device Chrome...                                       
30,211ms (!)
```
completes (or appears to) but nothing shown in browser window (White)

**Comment by DanTup** [comment]
I'm failing to reproduce this, though there seems to some variance in the reports above (whether it's at startup or hot reload, whether it happens on the sample counter project, and whether it happens only in VS Code versus at the terminal).

For anyone that can reproduce this (ideally from VS Code using the sample counter app), it may be useful to have a verbose log:

- Run the **Dart: Capture Logs** command in VS Code
- Untick "Analysis Server" but leave everything else ticked and click OK
- Run your project with `F5`
- Reproduce the issue (leave it stuck on "Syncing" for a few minutes, to ensure if there are any timeouts or similar, they're captured in the log)
- Click **Cancel* on the logging notification to stop logging and open the log
- Attach the log here
- Also open the Chrome DevTools and click on the Console tab, and copy any warnings/errors into here too

**Comment by alexda12** [comment]
@DanTup 

1. This is at Startup - running an EXISTING Flutter app via Run-Start Debugging
2. It does not happen on Counter App
3. It DOES not happen in terminal - but then in terminal, we get a BLANK screen

Note - I have managed to get past the 'Syncing files to device  Chrome - Stuck issue'

What I did : 

1. Create a brand new Flutter app and NAME it the same as an EXISTING app that you want to Run as WEB APP
2. Ensure you have Flutter config --enable-web it
3. Switch channels to latest Beta
4. Copy your LIB folder from existing project into this new Flutter app
5. Copy your pubspec from existing project into this new Flutter app
6. Update collection (if you are using this) to 1.14.12
7. Update your index.html accordingly

If you follow these steps - you will not be seeing the 'Syncing files to device chrome - stuck' issue.

However - I now have a new issue whereby I now get an exception : 

Launching lib/main.dart on Chrome in debug mode...


```
Debug service listening on ws://127.0.0.1:49678/ifXO0PZFa6g=


Error: Failed assertion: boolean expression must not be null
    at Object.throw_ [as throw] (http://localhost:49600/dart_sdk.js:4773:11)
    at Object.test (http://localhost:49600/dart_sdk.js:5058:32)
    at cloud_firestore_web.FirestoreWeb.new.settings (http://localhost:49600/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:774:18)
    at settings.next (<anonymous>)
    at runBody (http://localhost:49600/dart_sdk.js:35674:34)
    at Object._async [as async] (http://localhost:49600/dart_sdk.js:35702:7)
    at cloud_firestore_web.FirestoreWeb.new.settings (http://localhost:49600/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:767:20)
    at cloud_firestore.Firestore.new.settings (http://localhost:49600/packages/cloud_firestore/cloud_firestore.dart.lib.js:517:31)
    at main$ (http://localhost:49600/packages/PoL1/menu/test/test_start.dart.lib.js:224639:48)
    at main$.next (<anonymous>)
    at runBody (http://localhost:49600/dart_sdk.js:35674:34)
    at Object._async [as async] (http://localhost:49600/dart_sdk.js:35702:7)
    at Object.main$ [as main] (http://localhost:49600/packages/PoL1/menu/test/test_start.dart.lib.js:224635:18)
    at main$ (http://localhost:49600/web_entrypoint.dart.lib.js:17:12)
    at main$.next (<anonymous>)
    at onValue (http://localhost:49600/dart_sdk.js:35655:33)
    at _RootZone.runUnary (http://localhost:49600/dart_sdk.js:35540:58)
    at _FutureListener.thenAwait.handleValue (http://localhost:49600/dart_sdk.js:30956:29)
    at handleValueCallback (http://localhost:49600/dart_sdk.js:31466:49)
    at Function._propagateToListeners (http://localhost:49600/dart_sdk.js:31498:17)
    at async._AsyncCallbackEntry.new.callback (http://localhost:49600/dart_sdk.js:31251:27)
    at Object._microtaskLoop (http://localhost:49600/dart_sdk.js:35756:13)
    at _startMicrotaskLoop (http://localhost:49600/dart_sdk.js:35762:13)
    at http://localhost:49600/dart_sdk.js:31704:9[{"id":"1","result":{"value":"macOS","type":"","method":"ext.flutter.platformOverride"}}]

```

I am unsure why it mentions : PoL1/menu/test/test_start as I removed references to this and yet it still appears to pick up ... Anyway, anyone know where I can find generated JavaScript code for the dart classes so I can establish what dartjs doesnt like ?

**Comment by alexda12** [comment]
@DanTup 

I forgot to mention, when I turn on logging, I get these two exceptions thrown in Dart Logs : 

```
[1:21:30 PM] [Observatory] [Info] [Flutter (Chrome)] <== {"jsonrpc":"2.0","id":"6","error":{"code":-32603,"message":"UnimplementedError: The stream `Logging` is not supported.\n#0      ChromeProxyService.onEvent.<anonymous closure> (package:dwds/src/services/chrome_proxy_service.dart:515:11)\n#1      _LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:293:23)\n#2      ChromeProxyService.onEvent (package:dwds/src/services/chrome_proxy_service.dart:492:31)\n#3      VmServerConnection._delegateRequest (package:vm_service/src/vm_service.dart:1483:40)\n#4      _rootRunUnary (dart:async/zone.dart:1192:38)\n#5      _CustomZone.runUnary (dart:async/zone.dart:1085:19)\n#6      _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)\n#7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)\n#8      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)\n#9      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:134:11)\n#10     _MapStream._handleData (dart:async/stream_pipe.dart:234:10)\n#11     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:166:13)\n#12     _rootRunUnary (dart:async/zone.dart:1192:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1085:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)\n#17     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:779:19)\n#18     _StreamController._add (dart:async/stream_controller.dart:655:7)\n#19     _StreamController.add (dart:async/stream_controller.dart:597:5)\n#20     new WebSocketImpl.fromSocket.<anonymous closure> (package:web_socket_channel/src/copy/web_socket_impl.dart:719:21)\n#21     _rootRunUnary (dart:async/zone.dart:1192:38)\n#22     _CustomZone.runUnary (dart:async/zone.dart:1085:19)\n#23     _CustomZone.runUnaryGuarded (dart:async/zone.…
[1:21:30 PM] [Observatory] [Info] [Flutter (Chrome)] UnimplementedError: The stream `Logging` is not supported.
#0      ChromeProxyService.onEvent.<anonymous closure> (package:dwds/src/services/chrome_proxy_service.dart:515:11)
#1      _LinkedHashMapMixin.putIfAbsent (dart:collection-patch/compact_hash.dart:293:23)
#2      ChromeProxyService.onEvent (package:dwds/src/services/chrome_proxy_service.dart:492:31)
#3      VmServerConnection._delegateRequest (package:vm_service/src/vm_service.dart:1483:40)
#4      _rootRunUnary (dart:async/zone.dart:1192:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#6      _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#8      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#9      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:134:11)
#10     _MapStream._handleData (dart:async/stream_pipe.dart:234:10)
#11     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:166:13)
#12     _rootRunUnary (dart:async/zone.dart:1192:38)
#13     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#17     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:779:19)
#18     _StreamController._add (dart:async/stream_controller.dart:655:7)
#19     _StreamController.add (dart:async/stream_controller.dart:597:5)
#20     new WebSocketImpl.fromSocket.<anonymous closure> (package:web_socket_channel/src/copy/web_socket_impl.dart:719:21)
#21     _rootRunUnary (dart:async/zone.dart:1192:38)
#22     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#23     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#24     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:…
```

and 


```

[1:21:30 PM] [Observatory] [Info] [Flutter (Chrome)] UnimplementedError
#0      ChromeProxyService.setLibraryDebuggable (package:dwds/src/services/chrome_proxy_service.dart:566:5)
#1      VmServerConnection._delegateRequest (package:vm_service/src/vm_service.dart:1440:51)
#2      _rootRunUnary (dart:async/zone.dart:1192:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#6      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#7      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:134:11)
#8      _MapStream._handleData (dart:async/stream_pipe.dart:234:10)
#9      _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:166:13)
#10     _rootRunUnary (dart:async/zone.dart:1192:38)
#11     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#12     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#13     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#14     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#15     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:779:19)
#16     _StreamController._add (dart:async/stream_controller.dart:655:7)
#17     _StreamController.add (dart:async/stream_controller.dart:597:5)
#18     new WebSocketImpl.fromSocket.<anonymous closure> (package:web_socket_channel/src/copy/web_socket_impl.dart:719:21)
#19     _rootRunUnary (dart:async/zone.dart:1192:38)
#20     _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#21     _CustomZone.runUnaryGuarded (dart:async/zone.dart:987:7)
#22     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#23     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:266:7)
#24     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:70:11)
#25     _EventSinkWrapper.add (dart:async/stream_transf…
```

**Comment by DanTup** [comment]
@alexda12 

> This is at Startup - running an EXISTING Flutter app via Run-Start Debugging
> It does not happen on Counter App

If the issue doesn't occur in the counter app, but happens in your other app, it's likely there's some difference triggering this. I can't reproduce it any of the few apps I've tried, though the first error you posted looks specific to Firebase. Are you able to share a minimal project that reproduces this?

The other two errors you posted in the logs are known (things that aren't implemented in web yet) and shouldn't affect anything else, we just log them rather than silently skipping over them.

**Comment by KanybekMomukeyev** [comment]
Same here, VSCode is not working, but from terminal everything is ok.
This issue happened after updated to latest dev channel for mac.

**Comment by HvTechnologies** [comment]
I have solved it by downgrading the beta version to 1.17.

**Comment by DanTup** [comment]
Is anyone seeing this in VS Code able to share an exact project that triggers it (and exact hash of their Flutter version)? I'm currently unable to reproduce it and some comments above say it doesn't occur with the sample counter app so it may be specific to something in the app.

**Comment by trapeye** [comment]
I solve because my import C:/Users/user/Desktop/flutter/first_app/lib/src/screen.dart and I change it to import 'src/screen.dart';

**Comment by neil-dev** [comment]
Experiencing same issue.

**Comment by jasonminsookim** [comment]
Hey all. I had the same issue as everyone else when I upgraded my flutter. 

The fastest way to downgrade to a working version is to copy this into the vscode terminal.
`flutter downgrade `

**Comment by 4lfant** [comment]
Same issue here when running Flutter Web application. It appears on both `dev` and `beta` channels. 
About 65% of time the application get stuck on "Syncing files to device" and white screen in Chrome; 25% of time I get red screen with 
> TypeError cannot read property "%property%" of undefined

The property is different every time. It can be `ThemeData`, `HeroController` or any other items.
And only 10% of the time I get the working app. And yep, on average I must to run the app around 10 times to launch it successfully.

Running `flutter run -d chrome --verbose` from the terminal change nothing at all.
Also when I run the app from a terminal and it get stuck, the terminal stucks on 
> DevTools listening on ws://127.0.0.1:51912/devtools/browser/6b87c4a6-e0c9-45a1-bedf-aff9adb7bdfb

command.  When I open the url "127.0.0.1:51912" and choose second item "Service Worker" I get the scrren with the following error:
> Debugging connection was closed. Reason: WebSocket dissconected.
Recconect when ready by reopening DevTools.

When I lauch the app from a terminal successfully the output is following:

> [CHROME]:DevTools listening on ws://127.0.0.1:52848/devtools/browser/aff37e55-bd8a-4d31-855c-a6bda3a1d738
 Debug service listening on ws://127.0.0.1:52870/35DnZS9ywcQ=
 Debug service listening on ws://127.0.0.1:52870/35DnZS9ywcQ=
Warning: Flutter's support for web development is not stable yet and hasn't ...

Seems like the Debug service couldnt start in first scenario and the app get stuck because of it.

**Comment by 4lfant** [comment]
Nevermind. I just figured out I included the script for third-party library in my html body like that:
```
<body>
...
  <script src="main.dart.js" type="application/javascript"></script>
  <script src="mylibrary.js"></script>
</body>
```
It was the cause of a problem. I've just moved the script to the head tag and everything works fine:
```
<head>
...
  <script src="mylibrary.js"></script>
</head>
```

**Comment by AresVampire** [comment]
I am facing the same issue. Doesn't work on Android Studio / VSCode both.
I think it is related to `StreamProvider`.

**Comment by AresVampire** [comment]
![image](https://user-images.githubusercontent.com/26500585/87025157-06bdf400-c1a8-11ea-89a1-3a3d713dd844.png)

**Comment by uzhvij** [comment]
I have got this issue after applying firebase to the project and updating index.html and pubspec.yaml files.
When i have turned all back - issue disappeared. After that i have repeated all changes in the index.html file with slight differences and issue didn't come back.
All changes in the index.html file should be placed before this part:

<script>
    if ('serviceWorker' in navigator) {
      window.addEventListener('load', function () {
        navigator.serviceWorker.register('flutter_service_worker.js');
      });
    }
  </script>
  <script src="main.dart.js" type="application/javascript"></script>
</body>
</html>

It works for me.

**Comment by seanmcelroy** [comment]
> 
> 
> I have got this issue after applying firebase to the project and updating index.html and pubspec.yaml files.
> When i have turned all back - issue disappeared. After that i have repeated all changes with slight differences in the index.html file and issue didn't come back.
> All changes in the index.html file should be placed before this part:
> 
> !--<script>
> if ('serviceWorker' in navigator) {
> window.addEventListener('load', function () {
> navigator.serviceWorker.register('flutter_service_worker.js');
> });
> }
> </script>
> <script src="main.dart.js" type="application/javascript"></script> --!
> 
> It works for me.

I can confirm this exactly.  I started experiencing the issue after applying Firebase, but resolved it by ordering the <script> tags like:

1. CDN static links
2. Firebase config and init
3. Server worker stanza
4. main.dart.js

Thank you.

**Comment by uzhvij** [comment]
OK, I guess I put it wrong.
Here is my index file looks like. All works fine.

![index](https://user-images.githubusercontent.com/46085968/87248061-06e81900-c460-11ea-91ce-96237b176314.jpg)

**Comment by franarolas** [comment]
Ive encountered and solved this issue two times now:

The blank page happens when you have an absolute path to a file imported as a package, I know that no one would do it but I found out that Android Studio and Intellij idea sometimes makes this kind of mistake when you move a file to another path and it tries to refactor all the paths pointing to it, just check all imported packages and remove absolute paths.

I also got this problem because of the assets, I got a folder named "assets" with some subfolders like "images", "lang" etc, I was adding the assets folders paths into the pubspec.yaml but when I builded the project and I run it, it just got stuck, the build web was not correctly adding the assets to the build folder.
I just moved all the folders inside "assets" outside of it, modified pubspec and it worked like a charm.(Seems like it was internally encountering a problem with the folder name).

I hope my experiences can help some one.

**Comment by grouma** [comment]
@DanTup can you take a look at the issue outlined in https://github.com/flutter/flutter/issues/54290#issuecomment-668686132

I believe `package:dwds` and the corresponding asset server should handle absolute paths. I suspect there is something else going on in VS Code.

**Comment by DanTup** [comment]
@franarolas are you using VS Code? If so, could you try [capturing a log](https://dartcode.org/docs/logging/#capture-logs-command) (you can untick **Analysis Server**, but leave the others ticked)?

**Comment by DanTup** [comment]
@franarolas can you given an example of what you mean by absolute path? Do you mean a `package:` path, or a full absolute file path? When I try the latter I get errors both from the analyzer, and from Flutter when trying to run (see https://github.com/dart-lang/sdk/issues/42999).

```
Launching lib/main.dart on Chrome in debug mode...
Error: Invalid package URI 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart':
  Invalid argument (packageUri): Package URIs must not start with a '/': Instance of '_Uri'.
lib/main.dart:3:8: Error: Not found: 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart'
```

**Comment by franarolas** [comment]
> 
> 
> @franarolas can you given an example of what you mean by absolute path? Do you mean a `package:` path, or a full absolute file path? When I try the latter I get errors both from the analyzer, and from Flutter when trying to run (see [dart-lang/sdk#42999](https://github.com/dart-lang/sdk/issues/42999)).
> 
> ```
> Launching lib/main.dart on Chrome in debug mode...
> Error: Invalid package URI 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart':
>   Invalid argument (packageUri): Package URIs must not start with a '/': Instance of '_Uri'.
> lib/main.dart:3:8: Error: Not found: 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart'
> ```

Sorry for the delay.
I'm using IntelliJ IDEA right now, haven't worked with Flutter Web in VS Code yet.
What I mean as an absolute path is the following:
Flutter imports like that:
"import 'package:acat/tools/form_tools.dart';"
But sometimes with Intellij when you move the file(for example "form_tools.dart") it fails for some reason and it refactors like that:
"import 'C:\Users\F-Laptop\Documents\Projects\Flutter\acat\lib\newFolder\form_tools.dart';"

This kind of import makes the Flutter Web crash and show a white screen.
Hope it helps.

**Comment by franarolas** [comment]
> 
> 
> @franarolas can you given an example of what you mean by absolute path? Do you mean a `package:` path, or a full absolute file path? When I try the latter I get errors both from the analyzer, and from Flutter when trying to run (see [dart-lang/sdk#42999](https://github.com/dart-lang/sdk/issues/42999)).
> 
> ```
> Launching lib/main.dart on Chrome in debug mode...
> Error: Invalid package URI 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart':
>   Invalid argument (packageUri): Package URIs must not start with a '/': Instance of '_Uri'.
> lib/main.dart:3:8: Error: Not found: 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart'
> ```

So in your case just modify the import to:
import 'package:test_flutter_54290/danny.dart';
I hope it helps.

**Comment by DanTup** [comment]
> "import 'package:\Users\F-Laptop\Documents\Projects\Flutter\acat\lib\newFolder\form_tools.dart';"
> 
> This kind of import makes the Flutter Web crash and show a white screen.

When I tried to use a full path in a `package:` URI on macOS or Windows, the analyzer shows me that the code is invalid and Flutter won't let me launch the app (it gives the error I posted above).

<img width="826" alt="Screenshot 2020-08-31 at 17 14 03" src="https://user-images.githubusercontent.com/1078012/91742003-74d1d680-ebad-11ea-8358-411aa2b91a95.png">

Can you clarify exactly how you're reproducing this?

**Comment by devDeejay** [comment]
This checklist always helps me to fix this issue.

1. Make sure your app runs perfectly fine as an Android or iOS app. 
2. Make sure to check your web/index.html file to check for the syntax, all the <scripts> are there and functioning. 
3. Run the app and inspect your application for any errors in the console. 
4. If you are using Firebase, make sure you have imported all the required dependencies.

**Comment by franarolas** [comment]
> 
> 
> > "import 'package:\Users\F-Laptop\Documents\Projects\Flutter\acat\lib\newFolder\form_tools.dart';"
> > This kind of import makes the Flutter Web crash and show a white screen.
> 
> When I tried to use a full path in a `package:` URI on macOS or Windows, the analyzer shows me that the code is invalid and Flutter won't let me launch the app (it gives the error I posted above).
> 
> <img alt="Screenshot 2020-08-31 at 17 14 03" width="826" src="https://user-images.githubusercontent.com/1078012/91742003-74d1d680-ebad-11ea-8358-411aa2b91a95.png">
> 
> Can you clarify exactly how you're reproducing this?

In the case you are trying to run it in Windows it will obviusly fail, since the path "/Users/Danny/..." can not exist.
What is your folder structure and why would you like to import a package from an absolute path?

**Comment by DanTup** [comment]
> What is your folder structure and why would you like to import a package from an absolute path?

I was trying to reproduce what I understood the issue to be from above:

> But sometimes with Intellij when you move the file(for example "form_tools.dart") it fails for some reason and it refactors like that:
> "import 'package:\Users\F-Laptop\Documents\Projects\Flutter\acat\lib\newFolder\form_tools.dart';"

I tried both on Windows and macOS with various formats of paths but they all showed analysis issues, so I wasn't sure if I was reproducing the same thing.

**Comment by franarolas** [comment]
Ok I understand,
I will face this issue again in the next days 100%, I will post the issue when it happens.

**Comment by franarolas** [comment]
@DanTup I encountered this issue again, here are the details:

Intellij refactored the import of a package like that:
`import 'file:///C:/Users/F-Desktop/Documents/Projects/ACAT/acat_main/lib/pages/registration_page/forms/choose_user_type_form/choose_user_type_form.dart';`

It compiles but ends up in a white screen(Flutter Web).

`[√] Flutter (Channel beta, 1.21.0-9.2.pre, on Microsoft Windows [VersiÃ³n 10.0.18363.959], locale es-ES)
    • Flutter version 1.21.0-9.2.pre at C:\flutter
    • Framework revision 81a45ec2e5 (3 weeks ago), 2020-08-27 14:14:33 -0700
    • Engine revision 20a9531835
    • Dart version 2.10.0 (build 2.10.0-7.3.beta)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Users\F-Desktop\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] IntelliJ IDEA Ultimate Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2019.3.3
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361

[√] VS Code, 64-bit edition (version 1.45.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.10.2

[√] Connected device (3 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 85.0.4183.102
    • Edge (web)       • edge       • web-javascript • Microsoft Edge 85.0.564.51

• No issues found!
`

**Comment by DanTup** [comment]
@grouma I tried to reproduce the above on macOS and I also see a white screen (it works on iOS simulator), though it recorded a crash and generated the log below. I suspect it's the same issue.

I created a simple `flutter create` app then in `lib/main.dart` I just imported another file I created with an absolute path and `file://` scheme:

```
import 'file:///Users/danny/Desktop/TestProjects/abslute_path_test/lib/other.dart';
```

Then I ran it using the Chrome device (using VS Code, but the full command is below so could be reproduced from the terminal).

## command

flutter run --machine --target lib/main.dart -d chrome --track-widget-creation --dart-define=flutter.inspector.structuredErrors=true --start-paused --web-server-debug-protocol ws --web-server-debug-backend-protocol ws --web-allow-expose-url

## exception

ArgumentError: Invalid argument (URI scheme not allowed): "file:///Users/danny/Desktop/TestProjects/abslute_path_test/lib/other.dart"

```
#0      DartUri._recordAbsoluteUri (package:dwds/src/utilities/dart_uri.dart:86:7)
#1      DartUri.recordAbsoluteUris (package:dwds/src/utilities/dart_uri.dart:60:9)
<asynchronous suspension>
#2      AppInspector._initialize (package:dwds/src/debugging/inspector.dart:88:19)
<asynchronous suspension>
#3      AppInspector.initialize (package:dwds/src/debugging/inspector.dart:152:24)
#4      ChromeProxyService.createIsolate (package:dwds/src/services/chrome_proxy_service.dart:171:37)
<asynchronous suspension>
#5      ChromeProxyService.create (package:dwds/src/services/chrome_proxy_service.dart:138:23)
#6      DebugService.start (package:dwds/src/services/debug_service.dart:142:55)
#7      DevHandler._startLocalDebugService (package:dwds/src/handlers/dev_handler.dart:184:25)
#8      _rootRunUnary (dart:async/zone.dart:1198:47)
#9      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
#10     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
#11     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
#12     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
#13     Future._completeWithValue (dart:async/future_impl.dart:529:5)
#14     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
#15     _rootRun (dart:async/zone.dart:1190:13)
#16     _CustomZone.run (dart:async/zone.dart:1093:19)
#17     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
#18     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
#19     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#20     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#21     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#22     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

## flutter doctor

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.161, on Mac OS X 10.15.5 19F101, locale en-GB)
    • Flutter version 1.22.0-10.0.pre.161 at /Users/danny/Dev/Google/flutter
    • Framework revision cba91c97d0 (2 days ago), 2020-09-13 18:57:03 -0400
    • Engine revision 2fc054147e
    • Dart version 2.10.0 (build 2.10.0-121.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (4 available)
    • iPhone SE (2nd generation) (mobile) • DD509C0E-6A3D-4C85-B20F-6F0408238E4A • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)
    • macOS (desktop)                     • macos                                • darwin-x64     • Mac OS X 10.15.5 19F101
    • Web Server (web)                    • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 85.0.4183.102

• No issues found!
```

**Comment by grouma** [comment]
It seems there are multiple issues outlined in this thread.

**Absolute file URIs cause Flutter Tools / VS code to crash**

I discussed with the team. Generally users should not make use of absolute file URIs. They are not portable and have canonicalization issues. We will properly detect them in `package:dwds` to give a proper error in the short term. Long term, we will detect them in the compiler. Ideally we will have a corresponding lint as well. See the following issues:

https://github.com/dart-lang/webdev/issues/1125
https://github.com/dart-lang/sdk/issues/43639
https://github.com/dart-lang/linter/issues/2261

 
**Blank page in Chrome**

I have been able to reproduce this myself without the absolute file URI error. The root cause is still unknown so this will require further investigation.

**Comment by ductranit** [comment]
This issue happens on the latest master all the time.
I see the error `Uncaught TypeError for Web app with universal_io dependency in debug mode ` in chrome console log while the chrome tab is blank.
It is related to this issue https://github.com/flutter/flutter/issues/66859 
I notice that it works well when I create a new project or when I revert to older version on master

flutter doctor
```
[✓] Flutter (Channel unknown, 1.22.0-12.3.pre, on Mac OS X 10.15.6 19G2021, locale en-US) 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
[✓] Chrome - develop for the web
[!] Android Studio (version 4.0)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.49.2)
[✓] Connected device (3 available)

! Doctor found issues in 1 category.
```

**Comment by yasinarik** [comment]
I was trying to add a custom font which is `NewYorkExtraLarge-Medium.otf`. I created a `fonts`folder inside the `assets`folder. I added the appropriate changes into `pubspec.yaml`file. 

`flutter build web`works normally but the debug doesn't start.

After that, it got stuck. I removed and undo all the changes but it is still not working.

In the Chrome DevConsole it gives this error

```
Unhandled error detected in the injected client.js script.

You can disable this script in webdev by passing --no-injected-client if it
is preventing your app from loading, but note that this will also prevent
all debugging and hot reload/restart functionality from working.

The original error is below, please file an issue at
https://github.com/dart-lang/webdev/issues/new and attach this output:

NoSuchMethodError: method not found: 'get$digestsPath' (J.getInterceptor$x(...).get$digestsPath is not a function)
TypeError: J.getInterceptor$x(...).get$digestsPath is not a function
    at Object.get$digestsPath$x (http://localhost:52218/dwds/src/injected/client.js:3495:43)
```

**Comment by grouma** [comment]
@yasinarik I believe that's a dupe of https://github.com/dart-lang/webdev/issues/1133

I believe you simply need to update your SDK.

**Comment by yasinarik** [comment]
> @yasinarik I believe that's a dupe of [dart-lang/webdev#1133](https://github.com/dart-lang/webdev/issues/1133)
> 
> I believe you simply need to update your SDK.

I remembered that a few hours ago I used an app called `CleanMyMac`. It may have erased things but I am not sure.

1. I tried adding a version property. `<script src="main.dart.js" version= "1.0.1" type="application/javascript"></script>` It didn't work.

2. I added `dwds`dependency in `pubspec.yaml`file and run `flutter update-packages` command. It took a very long time but it didn't work at the end.
```
dependencies:
  dwds: ^6.0.0
```

3. Lastly, I basically upgraded the Flutter SDK by running `flutter upgrade`. 
Before: `1.22.0-2.0.pre.36`   -  Now: 1.23.0-19.0.pre.60`. I am using `master channel`

After the upgrade completed, I simply run `flutter clean` command to get rid of builds. Then closed the VSCode and reopened it. 

The debug on Chrome **now works again** as expected. Probably the step 3 is the only thing I should've done but I tried others too.

Thanks @grouma !

**Comment by antOnioOnio** [comment]
> > @franarolas can you given an example of what you mean by absolute path? Do you mean a `package:` path, or a full absolute file path? When I try the latter I get errors both from the analyzer, and from Flutter when trying to run (see [dart-lang/sdk#42999](https://github.com/dart-lang/sdk/issues/42999)).
> > ```
> > Launching lib/main.dart on Chrome in debug mode...
> > Error: Invalid package URI 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart':
> >   Invalid argument (packageUri): Package URIs must not start with a '/': Instance of '_Uri'.
> > lib/main.dart:3:8: Error: Not found: 'package:/Users/danny/Desktop/TestProjects/test_flutter_54290/lib/danny.dart'
> > ```
> 
> Sorry for the delay.
> I'm using IntelliJ IDEA right now, haven't worked with Flutter Web in VS Code yet.
> What I mean as an absolute path is the following:
> Flutter imports like that:
> "import 'package:acat/tools/form_tools.dart';"
> But sometimes with Intellij when you move the file(for example "form_tools.dart") it fails for some reason and it refactors like that:
> "import 'package:\Users\F-Laptop\Documents\Projects\Flutter\acat\lib\newFolder\form_tools.dart';"
> 
> This kind of import makes the Flutter Web crash and show a white screen.
> Hope it helps.

Thanks mate, you saved me some hours..

**Comment by ferhatb** [comment]
@grouma close ? any open issues?

**Comment by grouma** [comment]
We can probably close this one as we generally have better error messaging around these types of issues.
